### PR TITLE
feat: quest service auth and UI integration

### PR DESCRIPTION
**Pull request by AjimenezDCL**
## What does this PR change?

This PR adds the integration of quest service with the quests UI. It also adds the functionality for performing the auth to the quest server.
Quests are not yet completed but the PR will be sent in review to simplify the reviewing process considering that the code is excluded by a FF.

...

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at a383a49</samp>

This pull request adds a new RPC method `SignChallenge` to the sign request service and the decentraland protocol, which enables the authentication and authorization of the websocket connection with the quests service using the user identity. It also adds a new custom transport class `WebSocketClientWithChallengeTransport` that handles the challenge-response mechanism for the websocket connection. Additionally, it improves the quests service dumper script for testing purposes and fixes some formatting issues in the quality settings file.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+                {
+                    emptyState.SetActive(true);
+                    emptyActiveState.SetActive(false);
+                    emptyCompletedState.SetActive(true);
+                }
+                else
+                {
+                    emptyState.SetActive(false);
+                }
+            });
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs:90]: Can we extract those anons? just for clarity. `InProgressSectionOnSelect` and `CompletedSectionOnSelect` or whatever name fits 😄

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        }
+
+        private void ShowActiveOrCompletedQuests(bool active)
+        {
+            foreach (var activeQuest in activeQuests.Values)
+            {
+                activeQuest.gameObject.SetActive(active);
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs:97]: I didn't check the structure of the prefab, but if we have an extensive hierarchy of canvas objects, Toggling the Active state of game objects might lead to some hiccups, Is there a way we can toggle its canvas group alpha instead?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        }
+
+        public void AddActiveQuest(QuestDetailsComponentModel activeQuest)
+        {
+            emptyState.SetActive(false);
+
+            if (!activeQuests.ContainsKey(activeQuest.questId))
+                activeQuests.Add(activeQuest.questId, questsPool.Get());
+
+            activeQuests[activeQuest.questId].OnActiveQuestSelected -= SelectedQuest;
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs:112]: `activeQuests[activeQuest.questId]` is used 3 times, I would extract it into a local variable

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                completedQuest.gameObject.SetActive(!active);
+            }
+        }
+
+        public void AddActiveQuest(QuestDetailsComponentModel activeQuest)
+        {
+            emptyState.SetActive(false);
+
+            if (!activeQuests.ContainsKey(activeQuest.questId))
+                activeQuests.Add(activeQuest.questId, questsPool.Get());
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs:110]: We can simplify this with `TryAdd`

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs (hunk 3 above)
Comment (Kinerius) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs:112]: Same case at `AddCompletedQuest`

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestsController.cs [hunk 5]
```diff
@@ -1,91 +1,217 @@
...
+        private void AbandonQuest(string questId)
+        {
+            if(pinnedQuestId.Get().Equals(questId))
+                ChangePinnedQuest(questId, false);
+            questsService.AbortQuest(questId).Forget();
+            questLogComponentView.RemoveQuestIfExists(questId);
+        }
+
+        private void JumpIn(Vector2Int obj) =>
+            Environment.i.world.teleportController.Teleport(obj.x, obj.y);
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestsController.cs:83]: `Environment.i.world.teleportController` can be injected and mocked

**Review by Kinerius — CHANGES_REQUESTED** [review]
PR looking good, I have some suggestions

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs (hunk 2 above)
Comment (davidejensen) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Quests/QuestLogComponentView.cs:99]: The hierarchy for the quests list is not that complex

**Review by Kinerius — APPROVED** [review]
LGTM!

**Review by kwssbocian — APPROVED** [review]
✔️No issues observed for login flow and Explore menu appearance and functionality for both wallet and guest on WebGL and desktop.

**Pull request merged**